mantidproject/mantid
# Create Python StdoutChannel

Add a logging channel that go through python's wrapper of stdout.

**To test:**

In ipython notebook, run the following
```python
from mantid.kernel import ConfigService, Logger
Logger('a').notice('hi')  # will not appear with default channel
ConfigService.setString('logging.channels.consoleChannel.class', 'PythonStdoutChannel')
Logger('a').notice('bye')  # will appear in the notebook
```

Fixes #29987. 

#30118 is this into `ornl-next`

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
